.github/workflows/main.yml

```diff
@@ -77,3 +78,75 @@ jobs:
           npm run test-install
           # Undo those changes.
           git checkout --force
+
+  macos:
+    # https://github.com/actions/virtual-environments#available-environments
+    runs-on: macos-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+          ref: ${{ github.head_ref }}
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v1
+        with:
+          # Test only the oldest maintenance LTS Node.js version.
+          # https://github.com/nodejs/Release#release-schedule
+          node-version: 10
+
+      - name: Install dependencies
+        run: |
+          # Test platform-specific browser binary fetching for both
+          # Chromium and Firefox.
+          PUPPETEER_PRODUCT=firefox npm install
+          npm install
+          ls .local-chromium .local-firefox
+
+      - name: Build
+        run: |
+          npm run build
+
+      - name: Run unit tests
+        env:
+          CHROMIUM: true
+        run: |
+          npm run unit
+
+  windows:
+    # https://github.com/actions/virtual-environments#available-environments
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+          ref: ${{ github.head_ref }}
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v1
+        with:
+          # Test only the oldest maintenance LTS Node.js version.
+          # https://github.com/nodejs/Release#release-schedule
+          node-version: 10
+
+      - name: Install dependencies
+        run: |
+          # Test platform-specific browser binary fetching for both
+          # Chromium and Firefox.
+          $env:PUPPETEER_PRODUCT='firefox'
+          npm install
+          Remove-Item Env:\PUPPETEER_PRODUCT
```

Comment:
On the other platforms you're only running `npm install` once - is there a reason why windows is treated differently here?

Reply:
The intention is to run `npm install` twice (once for Firefox, once for Chromium), and AFAICT that’s what we‘re doing on all platforms — see lines 36-37 for Linux, 103-104 for macOS. What makes you say otherwise?

Reply:
Ah sorry, I missed the fact that the other platforms allow putting the `npm install` in the same line as  `PUPPETEER_PRODUCT=firefox`!